packages/xo-web/src/xo-app/vm/tab-advanced.js

```diff
@@ -588,6 +588,40 @@ export default class TabAdvanced extends Component {
     blockedOperations => STOP_OPERATIONS.every(op => op in blockedOperations)
   )
 
+  _getIsMigrationAllowed = createSelector(
+    () => this.props.vm?.blockedOperations,
+    blockedOperations =>
+      blockedOperations !== undefined && !['migrate_send', 'pool_migrate'].every(op => op in blockedOperations)
+  )
+
+  _onChangeAllowMigration = allow => {
+    const blockedOperations = this.props.vm.blockedOperations
+    const reasons = JSON.stringify(this.props.vm.blockedOperations)
+
+    const toggleBlockedOperations = () =>
+      editVm(this.props.vm, {
+        blockedOperations: Object.assign.apply(
+          null,
+          ['migrate_send', 'pool_migrate'].map(op => ({ [op]: allow ? null : 'Migration blocked' }))
+        ),
+      })
+
+    if (blockedOperations !== undefined && !['migrate_send', 'pool_migrate'].every(op => op in blockedOperations)) {
+      toggleBlockedOperations()
+    } else {
+      confirm({
+        title: _('allowMigrationTitle'),
+        body: (
+          <p>
+            {_('allowMigrationconfirm')} {reasons}
```

Comment:
This display is not good enough for users.

Do something like:

```js
<ul>
  {Object.keys(blockedOperations).map(op => {
    const reason = blockedOperations[op]
    if (op in MIGRATE_OPERATIONS && reason !== 'true') {
      return <li key={op}>reason</li>
    }
    return null
  })}
</ul>
```

Feel free to ask help from a front-end dev if needed.

Reply:
Done